core/src/test/java/org/elasticsearch/index/percolator/ExtractQueryTermsServiceTests.java

```diff
@@ -218,6 +220,31 @@ public void testExtractQueryMetadata_boostQuery() {
         assertThat(terms.get(0).bytes(), equalTo(termQuery1.getTerm().bytes()));
     }
 
+    public void testExtractQueryMetadata_commonTermsQuery() {
```

Comment:
And I want you to go through the builder just because I think tons of confusing stuff happens in there and its easy for things to look right at this level but not work properly in practice. I think the builders are more likely to catch things.